src/runtime/uv/dns.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+
+Author: Sofia Rodrigues, Henrik Böving
+*/
+#include "runtime/uv/dns.h"
+#include <cstring>
+
+namespace lean {
+#ifndef LEAN_EMSCRIPTEN
+
+using namespace std;
+
+// Std.Internal.IO.Async.DNS.getAddrInfo (host service : @& String) : IO (IO.Promise (Except IO.Error (Array SocketAddress)))
+extern "C" LEAN_EXPORT lean_obj_res lean_uv_dns_get_info(b_obj_arg name, b_obj_arg service, obj_arg /* w */) {
+    lean_object* promise = lean_promise_new();
+    mark_mt(promise);
+
+    uv_getaddrinfo_t* resolver = (uv_getaddrinfo_t*)malloc(sizeof(uv_getaddrinfo_t));
+    resolver->data = promise;
+
+    char const * name_cstr = lean_string_cstr(name);
+    char const * service_cstr = lean_string_cstr(service);
+
+    struct addrinfo hints;
+    memset(&hints, 0, sizeof(hints));
+    hints.ai_family = PF_UNSPEC;
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_protocol = IPPROTO_TCP;
```

Comment:
Are we sure we want to leave this completely unexposed? What if a user is e.g. looking for specifically an IPv4 or IPv6 address?